go/vt/vttablet/tabletserver/schema/historian.go

```diff
@@ -178,6 +182,13 @@ func (h *historian) loadFromDB(ctx context.Context) error {
 		h.lastID = id
 	}
 	h.sortSchemas()
+
+	loadedSchemas := len(h.schemas)
+	if loadedSchemas > maxVersions {
+		// keep only last N versions
+		h.schemas = h.schemas[loadedSchemas-maxVersions : loadedSchemas]
```

Comment:
@vmg If you have time to read this... can this cause a memory leak?

Reply:
I'm pretty sure that isn't saving memory. It's still a slice on the same backing array and this is just moving the window.

Reply:
From the article linked the fix is to
```
 is to nil out the trailing pointer(s) before you truncate the slice.
```

Reply:
@derekperkins Is the new version acceptable?